GroceryStore/ProductDetailViewController.m

```diff
@@ -0,0 +1,123 @@
+}
+
+- (IBAction)removeAllStock:(id)sender {
+    
+    //create session
+    [self createSessionWithURLComponent:@"inventory"];
+    [self createURLRequestWithHTTPMethod:@"DELETE"];
+    
+    //create task
+    NSURLSessionDataTask *deleteProduct = [self.session dataTaskWithRequest:self.URLRequest completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+        
+        [self createAndHandleURLResponse:response withData:data];
+    }];
+    
+    //resume
+    [deleteProduct resume];
+}
+
+- (IBAction)restockProduct:(id)sender { //add
+    //TODO? Can't stock product not currently/previously in inventory (201)
+    [self createSessionWithURLComponent:@"inventory"];
+    [self createURLRequestWithHTTPMethod:@"POST"];
+    [self implementProductAdjustment];
+    
+}
+
+- (IBAction)purchaseProduct:(id)sender { //subtract
+    
+    [self createSessionWithURLComponent:@"purchase"];
+    [self createURLRequestWithHTTPMethod:@"POST"];
+    [self implementProductAdjustment];
+}
+
+- (void)createSessionWithURLComponent:(NSString *)URLComponent {
+    
+    //create session
+    self.URLString = [NSString stringWithFormat:@"http://127.0.0.1:5000/api/%@/%@", URLComponent, self.productName];
+    
+    NSURLSessionConfiguration *sessionConfig = [NSURLSessionConfiguration defaultSessionConfiguration];
+    
+    self.session = [NSURLSession sessionWithConfiguration:sessionConfig];
+    
+}
+
+- (void)createURLRequestWithHTTPMethod:(NSString *)method {
+    
+    //URL request
+    self.URLRequest = [[NSMutableURLRequest alloc] initWithURL:[NSURL URLWithString:self.URLString]];
+    self.URLRequest.HTTPMethod = method;
+    [self.URLRequest setValue:@"application/json" forHTTPHeaderField:@"Content-type"];
+}
+
+- (void)createAndHandleURLResponse:(NSURLResponse *)response withData:(NSData *)data {
+    
+    NSHTTPURLResponse *httpResp =
+    (NSHTTPURLResponse *) response;
+    if (httpResp.statusCode == 200) {
+        NSError *jsonError;
+        NSDictionary *jsonDict = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingAllowFragments error:&jsonError];
+        
+        if (!jsonError) {
+            NSLog(@"%@", jsonDict);
+            dispatch_async(dispatch_get_main_queue(), ^{
+                
+                //update UI
+                self.productQuantityLabel.text = [NSString stringWithFormat:@"Current quantity is: %@", jsonDict[self.productName]];
+                
+            });
+        }
+    }
+}
+
+- (void)implementProductAdjustment {
+    
+    //specify amount to add
+    NSDictionary *quantityDict = @{@"quantity" : @(self.stepperOutlet.value)};
+    
+    NSData *quantityToPass = [NSJSONSerialization dataWithJSONObject:quantityDict options:NSJSONWritingPrettyPrinted error:nil];
+    
+    //create task
+    NSURLSessionUploadTask *restock = [self.session uploadTaskWithRequest:self.URLRequest fromData:quantityToPass completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
```

Comment:
uploadTask has uploadTaskWithRequest fromData parameter. not seeing an equivalent method for NSURLSessionDataTask that would take NSData to pass that information. Think i have to use UploadTask